File: src/java/SQ_Energy_Balance/EnergybalanceRate.java
import  java.io.*;
import  java.util.*;
import java.time.LocalDateTime;

public class EnergyBalanceRate
{
    private Double evapoTranspirationPriestlyTaylor;
    private Double evapoTranspirationPenman;
    private Double evapoTranspiration;
    private Double potentialTranspiration;
    private Double soilHeatFlux;
    private Double cropHeatFlux;
    
    public EnergyBalanceRate() { }
    
    public EnergyBalanceRate(EnergyBalanceRate toCopy, boolean copyAll) // copy constructor 
    {
        if (copyAll)
        {
            this.evapoTranspirationPriestlyTaylor = toCopy.getevapoTranspirationPriestlyTaylor();
            this.evapoTranspirationPenman = toCopy.getevapoTranspirationPenman();
            this.evapoTranspiration = toCopy.getevapoTranspiration();
            this.potentialTranspiration = toCopy.getpotentialTranspiration();
            this.soilHeatFlux = toCopy.getsoilHeatFlux();
            this.cropHeatFlux = toCopy.getcropHeatFlux();
            this.evapoTranspirationPriestlyTaylor = toCopy.getevapoTranspirationPriestlyTaylor();
            this.evapoTranspirationPenman = toCopy.getevapoTranspirationPenman();
            this.evapoTranspiration = toCopy.getevapoTranspiration();
            this.potentialTranspiration = toCopy.getpotentialTranspiration();
            this.soilHeatFlux = toCopy.getsoilHeatFlux();
            this.cropHeatFlux = toCopy.getcropHeatFlux();
        }
    }
    public Double getevapoTranspirationPriestlyTaylor()
    { return evapoTranspirationPriestlyTaylor; }

    public void setevapoTranspirationPriestlyTaylor(Double _evapoTranspirationPriestlyTaylor)
    { this.evapoTranspirationPriestlyTaylor= _evapoTranspirationPriestlyTaylor; } 
    
    public Double getevapoTranspirationPenman()
    { return evapoTranspirationPenman; }

    public void setevapoTranspirationPenman(Double _evapoTranspirationPenman)
    { this.evapoTranspirationPenman= _evapoTranspirationPenman; } 
    
    public Double getevapoTranspiration()
    { return evapoTranspiration; }

    public void setevapoTranspiration(Double _evapoTranspiration)
    { this.evapoTranspiration= _evapoTranspiration; } 
    
    public Double getpotentialTranspiration()
    { return potentialTranspiration; }

    public void setpotentialTranspiration(Double _potentialTranspiration)
    { this.potentialTranspiration= _potentialTranspiration; } 
    
    public Double getsoilHeatFlux()
    { return soilHeatFlux; }

    public void setsoilHeatFlux(Double _soilHeatFlux)
    { this.soilHeatFlux= _soilHeatFlux; } 
    
    public Double getcropHeatFlux()
    { return cropHeatFlux; }

    public void setcropHeatFlux(Double _cropHeatFlux)
    { this.cropHeatFlux= _cropHeatFlux; } 
    
}

File: src/cs/SQ_Energy_Balance/EnergyBalanceRate.cs
using System;
using System.Collections.Generic;

public class EnergyBalanceRate 
{
    private double _evapoTranspirationPriestlyTaylor;
    private double _evapoTranspirationPenman;
    private double _evapoTranspiration;
    private double _potentialTranspiration;
    private double _soilHeatFlux;
    private double _cropHeatFlux;
    
        public EnergyBalanceRate() { }
    
    
    public EnergyBalanceRate(EnergyBalanceRate toCopy, bool copyAll) // copy constructor 
    {
    if (copyAll)
    {
    
    evapoTranspirationPriestlyTaylor = toCopy.evapoTranspirationPriestlyTaylor;
    evapoTranspirationPenman = toCopy.evapoTranspirationPenman;
    evapoTranspiration = toCopy.evapoTranspiration;
    potentialTranspiration = toCopy.potentialTranspiration;
    soilHeatFlux = toCopy.soilHeatFlux;
    cropHeatFlux = toCopy.cropHeatFlux;
    }
    }
    public double evapoTranspirationPriestlyTaylor
        {
            get { return this._evapoTranspirationPriestlyTaylor; }
            set { this._evapoTranspirationPriestlyTaylor= value; } 
        }
    public double evapoTranspirationPenman
        {
            get { return this._evapoTranspirationPenman; }
            set { this._evapoTranspirationPenman= value; } 
        }
    public double evapoTranspiration
        {
            get { return this._evapoTranspiration; }
            set { this._evapoTranspiration= value; } 
        }
    public double potentialTranspiration
        {
            get { return this._potentialTranspiration; }
            set { this._potentialTranspiration= value; } 
        }
    public double soilHeatFlux
        {
            get { return this._soilHeatFlux; }
            set { this._soilHeatFlux= value; } 
        }
    public double cropHeatFlux
        {
            get { return this._cropHeatFlux; }
            set { this._cropHeatFlux= value; } 
        }
}